web3.js/src/connection.ts

```diff
@@ -955,23 +955,21 @@ function createRpcClient(
   url: string,
   useHttps: boolean,
   httpHeaders?: HttpHeaders,
-  customFetch?: typeof crossFetch,
+  customFetch?: FetchFn,
   fetchMiddleware?: FetchMiddleware,
   disableRetryOnRateLimit?: boolean,
 ): RpcClient {
-  const fetch = customFetch ? customFetch : crossFetch;
+  const fetchImpl = (customFetch ? customFetch : fetch) as FetchFn;
```

Comment:
nit: how about importing `./fetch-iml` as `fetchImpl` and keeping this as `fetch`?